activerecord/lib/active_record/connection_adapters/sqlite3/schema_dumper.rb

```diff
@@ -0,0 +1,13 @@
+module ActiveRecord
+  module ConnectionAdapters
+    module SQLite3
+      module ColumnDumper
+        private
+
+          def default_primary_key?(column)
+            schema_type(column) == :integer
+          end
```

Comment:
@matthewd @sgrif Trying to follow... what needs to change exactly? I'm reading conflicting opinions and want to get it right.